cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1874,6 +1879,32 @@ String zkConnectionString(int connectToReplicas, Function<Integer, String> zkNod
             return withVoid(fut);
         }
 
+        /**
+         * Get the cluster Id of the Kafka cluster
+         * 
+         * @return
+         */
+        Future<ReconciliationState> kafkaGetClusterId() {
+            String clusterId = null;
+            log.debug("{}: Attempt to get clusterId", reconciliation);
+            List<String> podNames = IntStream.range(0, kafkaCluster.getReplicas()).boxed().collect(Collectors.toList()).stream().map(podId -> KafkaCluster.kafkaPodName(this.name, podId)).collect(Collectors.toList());
+            try {
+                String bootstrapHostnames = podNames.stream().map(podName -> KafkaCluster.podDnsName(this.namespace, this.name, podName) + ":" + KafkaCluster.REPLICATION_PORT).collect(Collectors.joining(","));
+                log.debug("{}: Creating AdminClient for clusterId using {}", reconciliation, bootstrapHostnames);
+                Admin kafkaAdmin = adminClientProvider.createAdminClient(bootstrapHostnames, this.clusterCa.caCertSecret(), this.oldCoSecret, "cluster-operator");
```

Comment:
Is `oldCoSecret` really the right one to use? If this is old, it sounds like sometimes it might not be valid anymore.